internal/plugin_test.go

```diff
@@ -2467,3 +2467,249 @@ func TestCreatePolicyWithNamespaceSelector(t *testing.T) {
 		})
 	}
 }
+
+func TestGenerateNonDNSPolicyName(t *testing.T) {
+	t.Parallel()
+	tmpDir := t.TempDir()
+	createConfigMap(t, tmpDir, "configmap.yaml")
+
+	p := Plugin{}
+	var err error
+
+	p.baseDirectory, err = filepath.EvalSymlinks(tmpDir)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	p.PlacementBindingDefaults.Name = "my-placement-binding"
+	p.PolicyDefaults.Placement.Name = "my-placement-rule"
+	p.PolicyDefaults.Namespace = "my-policies"
+	capitalizedName := "policy-APP-CONFIG"
+	policyConf := types.PolicyConfig{
+		Name: capitalizedName,
+		Manifests: []types.Manifest{
+			{Path: path.Join(tmpDir, "configmap.yaml")},
+		},
+	}
+	basePolicies := p.Policies
+	p.Policies = append(basePolicies, policyConf)
+	p.applyDefaults(map[string]interface{}{})
+
+	if err := p.assertValidConfig(); err != nil {
+		t.Fatal(err.Error())
+	}
+
+	_, err = p.Generate()
```

Comment:
Could you make each test in these tests a subtest?

The benefits are:

- It's clearer which test data caused the failure
- The sub tests can run in parallel
- I think it also makes it so that one test failure does not cause the rest of the tests to not run
- We get individual reporting per test case

For example:
https://github.com/stolostron/policy-generator-plugin/blob/49e1893da8f8fc0e1b3c93c3b17b093f3a58aaae/internal/plugin_test.go#L2296-L2316